pkg/issuer/util.go

```diff
@@ -0,0 +1,25 @@
+package issuer
+
+import (
+	"fmt"
+
+	"github.com/jetstack/cert-manager/pkg/apis/certmanager/v1alpha1"
+)
+
+func ValidateDuration(issuer v1alpha1.GenericIssuer) error {
+	if issuer.GetSpec().Duration != 0 && issuer.GetSpec().RenewBefore == 0 &&
+		issuer.GetSpec().Duration <= MinimumCertificateDuration {
+		return fmt.Errorf("certificate duration must be greater than 30 days if spec.renewBefore is not set (%s)", MinimumCertificateDuration)
+	}
+
+	if issuer.GetSpec().Duration != 0 && issuer.GetSpec().RenewBefore != 0 &&
+		issuer.GetSpec().Duration < issuer.GetSpec().RenewBefore {
+		return fmt.Errorf("certificate duration (%s) must be greater than renewBefore (%s)", issuer.GetSpec().Duration, issuer.GetSpec().RenewBefore)
```

Comment:
The idea here is that if the configuration entered by the user is already known to be an error an error is generated. If it is because of the duration received by the certificate server (ACME, Vault etc) then the 1/3 rule is used and a warning is generated. We could also use the same rule here though if you think it's best.

Reply:
👍 that sounds okay to me.

Perhaps in future the Issuers controller will handle setting the ready state of the issues to 'false' if its own configuration is invalid (and/or we outright reject invalid issuer specs using a ValidatingAdmissionWebhook)